views/AnimatedProfile.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react'
+import { View, Text, Image, StyleSheet, TextInput, Dimensions, Animated, Easing, TouchableOpacity } from "react-native"
+import { connect } from "react-redux"
+import Badges from "../components/Badges"
+
+const { height, width } = Dimensions.get("window")
+
+interface Props {
+  username: string,
+  showProfile: boolean,
+  setUserName: Function,
+  closeProfile: Function
+  currentProfile: User
+}
+
+interface User {
+  first_name: string,
+  profile_photo: string,
+  lvl: string
+}
+
+export class AnimatedProfile extends Component<Props> {
+  componentDidUpdate() {
```

Comment:
componentDidUpdate controls if the component should be animated in or out, based on the value of "showProfile"